File: 111/InsertionLinkedList.py
def main():
    n = int(input())
    head = None
    for i in range(n):
        k=int(input())
        if(head==None):
            head = Node(k)
        else:
            insertAtEnd(head,k)

    iter = head
    while(iter!=None):
        print(iter.val,end=" ")
        iter=iter.next

class Node:
    def __init__(self,val: int):
        self.next = None
        self.val = val

def insertAtEnd(head: Node, n:int):
    current = head
    while(current.next!=None):
        current=current.next
    current.next = Node(n)

if __name__ == "__main__":
    main()

File: 111/InsertionLinkedList.dart
import 'dart:io';

class Node {
  int val;
  Node? next;

  Node(this.val);
}

void insertAtEnd(Node head, int n) {
  Node current = head;

  while (current.next != null) {
    current = current.next!;
  }

  current.next = Node(n);
}

void main() {
  int n = int.parse(stdin.readLineSync()!);

  Node? head;

  for (int i = 0; i < n; i++) {
    int k = int.parse(stdin.readLineSync()!);

    if (head == null) {
      head = Node(k);
    } else {
      insertAtEnd(head, k);
    }
  }

  Node? iter = head;

  while (iter != null) {
    stdout.write("${iter.val} ");
    iter = iter.next;
  }
}
